src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -429,6 +434,123 @@ GetShardCost(uint64 shardId, void *voidContext)
 }
 
 
+/*
+ * worker_partitioned_relation_total_size returns the total size (including indexes) of
+ * a partitioned relation with its partitions, using the function pg_partition_tree.
+ */
+Datum
+worker_partitioned_relation_total_size(PG_FUNCTION_ARGS)
+{
+	char *relname = text_to_cstring(PG_GETARG_TEXT_P(0));
+
+	/* pg_partition_tree finds the partitions recursively and returns them */
+	char *queryText = "SELECT sum(pg_total_relation_size(relid))::bigint AS total_size "
+					  "FROM (SELECT relid from pg_partition_tree($1)) partition_tree;";
+
+	uint64 size = ExecSizeQueryWithRelName(queryText, relname);
+
+	PG_RETURN_UINT64(size);
```

Comment:
glancing the sql definitions we do return a `bigint` same type as we are casting the result of the sum to. Why is there a difference between reading the number from the SPI as `Int64` and returning an `UINT64`?

Would be great to have the same types everywhere as the original functions have, baring enough space to express the sum of sizes in.